### DISCUSSION TITLE
Privoxy instability 404 127.0.0.1:8118

### DISCUSSION THREAD

**Discussion by schonabe** [PurpleI2P/i2pd#1838] [Q&A]
It works without a proxy but I don't trust like that ![Untitled](https://user-images.githubusercontent.com/119639620/210458313-77703d38-632f-492a-9506-72afa5f1954f.png)
I have tried config.txt but 7070 or 4444 are the only settings that don't produce an error on boot "accessing socket forbidden by access permissions". my firewall all checks out, maybe i need to restart my pc? Privoxy works on clearnet.

**Comment by r4sas** [comment]
Show your privoxy configuration.

**Comment by schonabe** [comment]
This error is happening since restarting my PC (attached photo). I added logs in config but nothing else. My listen address is: 127.0.0.1:8118 but i didn't set that manually. Thanks

> too long, stripped -- R4SAS

**Comment by schonabe** [comment]
The error is: "fatal error: cant bind to 127.0.0.1:8118: there may be another privoxy or some other proxy running on port 8118"
after it loads filter and action files. Thanks.






On Thursday, 5 January 2023 at 12:39:47 am AEDT, R4SAS ***@***.***> wrote: 







Show your privoxy configuration.

**Reply by r4sas** [reply]
This is not i2pd issue. Please refer to privoxy documentation.

**Comment by schonabe** [comment]
The Privoxy service didn't start automatically (so immature of me). 
I2dp still has "unable to start web console and proxy at 127.0.0.1:8118" OR it wont load i2d pages when the proxy is set in firefox (portable).
I changed http back to 4444 and i received and SSU error, but that disappeared when i changed proxy back to 4444 and also when i tried to recreate the error. If you can help it would be nice.

below I changed http and proxy to 8118 but neither worked set to 7070 or 4444.

## Configuration file for a typical i2pd user
## See https://i2pd.readthedocs.io/en/latest/user-guide/configuration/
## for more options you can use in this file.


## Lines that begin with "## " try to explain what's going on. Lines
## that begin with just "#" are disabled commands: you can enable them
## by removing the "#" symbol.


## Tunnels config file
## Default: ~/.i2pd/tunnels.conf or /var/lib/i2pd/tunnels.conf
# tunconf = /var/lib/i2pd/tunnels.conf


## Tunnels config files path
## Use that path to store separated tunnels in different config files.
## Default: ~/.i2pd/tunnels.d or /var/lib/i2pd/tunnels.d
# tunnelsdir = /var/lib/i2pd/tunnels.d


## Path to certificates used for verifying .su3, families
## Default: ~/.i2pd/certificates or /var/lib/i2pd/certificates
# certsdir = /var/lib/i2pd/certificates


## Where to write pidfile (default: i2pd.pid, not used in Windows)
# pidfile = /run/i2pd.pid


## Logging configuration section
## By default logs go to stdout with level 'info' and higher
## For Windows OS by default logs go to file with level 'warn' and higher
##
## Logs destination (valid values: stdout, file, syslog)
##  * stdout - print log entries to stdout
##  * file - log entries to a file
##  * syslog - use syslog, see man 3 syslog
# log = file
## Path to logfile (default - autodetect)
# logfile = /var/log/i2pd/i2pd.log
## Log messages above this level (debug, info, *warn, error, none)
## If you set it to none, logging will be disabled
# loglevel = warn
## Write full CLF-formatted date and time to log (default: write only time)
# logclftime = true


## Daemon mode. Router will go to background after start. Ignored on Windows
# daemon = true


## Specify a family, router belongs to (default - none)
# family =


## Network interface to bind to
## Updates address4/6 options if they are not set
# ifname =
## You can specify different interfaces for IPv4 and IPv6
# ifname4 =
# ifname6 =


## Local address to bind transport sockets to
## Overrides host option if:
## For ipv4: if ipv4 = true and nat = false
## For ipv6: if 'host' is not set or ipv4 = true
# address4 =
# address6 =


## External IPv4 or IPv6 address to listen for connections
## By default i2pd sets IP automatically
## Sets published NTCP2v4/SSUv4 address to 'host' value if nat = true
## Sets published NTCP2v6/SSUv6 address to 'host' value if ipv4 = false
# host = 1.2.3.4


## Port to listen for connections
## By default i2pd picks random port. You MUST pick a random number too,
## don't just uncomment this
# port = 4567


## Enable communication through ipv4
ipv4 = true
## Enable communication through ipv6
ipv6 = false


## Enable SSU transport
ssu = false


## Bandwidth configuration
## L limit bandwidth to 32KBs/sec, O - to 256KBs/sec, P - to 2048KBs/sec,
## X - unlimited
## Default is L (regular node) and X if floodfill mode enabled. If you want to
## share more bandwidth without floodfill mode, uncomment that line and adjust
## value to your possibilities
# bandwidth = L
## Max % of bandwidth limit for transit. 0-100. 100 by default
# share = 100


## Router will not accept transit tunnels, disabling transit traffic completely
## (default = false)
# notransit = true


## Router will be floodfill
## Note: that mode uses much more network connections and CPU!
# floodfill = true


[ntcp2]
## Enable NTCP2 transport (default = true)
# enabled = true
## Publish address in RouterInfo (default = true)
# published = true
## Port for incoming connections (default is global port option value)
# port = 4567


[ssu2]
## Enable SSU2 transport
# enabled = true
## Publish address in RouterInfo
# published = true
## Port for incoming connections (default is global port option value or port + 1 if SSU is enabled)
# port = 4567


[http]
## Web Console settings
## Uncomment and set to 'false' to disable Web Console
# enabled = true
## Address and port service will listen on
address = 127.0.0.1
port = 8118
## Path to web console, default "/"
# webroot = /
## Uncomment following lines to enable Web Console authentication
# auth = true
# user = i2pd
# pass = changeme
## Select webconsole language
## Currently supported english (default), afrikaans, armenian, chinese, french,
## german, italian, russian, spanish, turkmen, ukrainian and uzbek languages
# lang = english


[httpproxy]
## Uncomment and set to 'false' to disable HTTP Proxy
# enabled = true
## Address and port service will listen on
address = 127.0.0.1
port = 8118
## Optional keys file for proxy local destination
# keys = http-proxy-keys.dat
## Enable address helper for adding .i2p domains with "jump URLs" (default: true)
# addresshelper = true
## Address of a proxy server inside I2P, which is used to visit regular Internet
# outproxy = http://false.i2p
## httpproxy section also accepts I2CP parameters, like "inbound.length" etc.


[socksproxy]
## Uncomment and set to 'false' to disable SOCKS Proxy
# enabled = true
## Address and port service will listen on
address = 127.0.0.1
port = 4447
## Optional keys file for proxy local destination
# keys = socks-proxy-keys.dat
## Socks outproxy. Example below is set to use Tor for all connections except i2p
## Uncomment and set to 'true' to enable using of SOCKS outproxy
# outproxy.enabled = false
## Address and port of outproxy
# outproxy = 127.0.0.1
# outproxyport = 9050
## socksproxy section also accepts I2CP parameters, like "inbound.length" etc.


[sam]
## Comment or set to 'false' to disable SAM Bridge
enabled = true
## Address and port service will listen on
# address = 127.0.0.1
# port = 7656


[bob]
## Uncomment and set to 'true' to enable BOB command channel
# enabled = false
## Address and port service will listen on
# address = 127.0.0.1
# port = 2827


[i2cp]
## Uncomment and set to 'true' to enable I2CP protocol
# enabled = false
## Address and port service will listen on
# address = 127.0.0.1
# port = 7654


[i2pcontrol]
## Uncomment and set to 'true' to enable I2PControl protocol
# enabled = false
## Address and port service will listen on
# address = 127.0.0.1
# port = 7650
## Authentication password. "itoopie" by default
# password = itoopie


[precomputation]
## Enable or disable elgamal precomputation table
## By default, enabled on i386 hosts
# elgamal = true


[upnp]
## Enable or disable UPnP: automatic port forwarding (enabled by default in WINDOWS, ANDROID)
# enabled = false
## Name i2pd appears in UPnP forwardings list (default = I2Pd)
# name = I2Pd


[meshnets]
## Enable connectivity over the Yggdrasil network
# yggdrasil = false
## You can bind address from your Yggdrasil subnet 300::/64
## The address must first be added to the network interface
# yggaddress =


[reseed]
## Options for bootstrapping into I2P network, aka reseeding
## Enable or disable reseed data verification.
verify = true
## URLs to request reseed data from, separated by comma
## Default: "mainline" I2P Network reseeds
# urls = https://reseed.i2p-projekt.de/,https://i2p.mooo.com/netDb/,https://netdb.i2p2.no/
## Reseed URLs through the Yggdrasil, separated by comma
# yggurls = http://[324:9de3:fea4:f6ac::ace]:7070/
## Path to local reseed data file (.su3) for manual reseeding
# file = /path/to/i2pseeds.su3
## or HTTPS URL to reseed from
# file = https://legit-website.com/i2pseeds.su3
## Path to local ZIP file or HTTPS URL to reseed from
# zipfile = /path/to/netDb.zip
## If you run i2pd behind a proxy server, set proxy server for reseeding here
## Should be http://address:port or socks://address:port
# proxy = http://127.0.0.1:8118
## Minimum number of known routers, below which i2pd triggers reseeding. 25 by default
# threshold = 25


[addressbook]
## AddressBook subscription URL for initial setup
## Default: reg.i2p at "mainline" I2P Network
# defaulturl = http://shx5vqsw7usdaunyzr2qmes2fq37oumybpudrd4jjj4e4vk4uusa.b32.i2p/hosts.txt
## Optional subscriptions URLs, separated by comma
# subscriptions = http://reg.i2p/hosts.txt,http://identiguy.i2p/hosts.txt,http://stats.i2p/cgi-bin/newhosts.txt,http://rus.i2p/hosts.txt


[limits]
## Maximum active transit sessions (default:2500)
# transittunnels = 2500
## Limit number of open file descriptors (0 - use system limit)
# openfiles = 0
## Maximum size of corefile in Kb (0 - use system limit)
# coresize = 0


[trust]
## Enable explicit trust options. false by default
# enabled = true
## Make direct I2P connections only to routers in specified Family.
# family = MyFamily
## Make direct I2P connections only to routers specified here. Comma separated list of base64 identities.
# routers =
## Should we hide our router from other routers? false by default
# hidden = true


[exploratory]
## Exploratory tunnels settings with default values
# inbound.length = 2
# inbound.quantity = 3
# outbound.length = 2
# outbound.quantity = 3


[persist]
## Save peer profiles on disk (default: true)
# profiles = true
## Save full addresses on disk (default: true)
# addressbook = true


[cpuext]
## Use CPU AES-NI instructions set when work with cryptography when available (default: true)
# aesni = true
## Use CPU AVX instructions set when work with cryptography when available (default: true)
# avx = true
## Force usage of CPU instructions set, even if they not found
## DO NOT TOUCH that option if you really don't know what are you doing!
# force = false








On Friday, 6 January 2023 at 05:28:10 am AEDT, R4SAS ***@***.***> wrote: 







This is not i2pd issue. Please refer to privoxy documentation.

**Reply by r4sas** [reply]
Tell me, what you want to gather? You want your privoxy be able to go to i2p sites?

**Comment by schonabe** [comment]
I want to use it as intended, Privoxy as a proxy to mask my identity while i use i2pd is how i assume it works.
     On Friday, 6 January 2023, 07:52:59 am AEDT, R4SAS ***@***.***> wrote:  
 
 


Tell me, what you want to gather? You want your privoxy be able to go to i2p sites?

**Reply by r4sas** [reply]
Privoxy is not masking your identity. Use preconfigured Firefox for using I2P in that case: https://github.com/PurpleI2P/i2pdbrowser

**Comment by schonabe** [comment]
how is i2p over firefox (i2pd?) masing my identity more than a proxy?
     On Friday, 6 January 2023, 11:18:04 am AEDT, R4SAS ***@***.***> wrote:  
 
 


Privoxy is not masking your identity. Use preconfigured Firefox for using I2P in that case: https://github.com/PurpleI2P/i2pdbrowser

**Reply by r4sas** [reply]
This is preconfigured Firefox, not default one.
